GOOD-I-DEER/nodes/cosine-similarity-node/calculate-cosine.js

```diff
@@ -30,52 +36,53 @@ module.exports = function (RED) {
         magnitudeB = Math.sqrt(magnitudeB);
 
         if (magnitudeA === 0 || magnitudeB === 0) {
-          return false;
+          return ERROR_VECTOR_LENGTH_ZERO;
         }
 
         const similarity = dotProduct / (magnitudeA * magnitudeB);
+        if (isNaN(similarity)) return ERROR_COSINE_SIMILARITY_NAN;
         resultArray.push(similarity);
       }
       return resultArray;
     }
 
     async function getStoredVector() {
-      var stored = [[0]];
-
+      let stored = [[0]];
       const fs = require("fs");
       const filePath = config.file;
-
       try {
         const data = await fs.promises.readFile(filePath, "utf8");
         stored = JSON.parse(data);
       } catch (err) {
-        console.error("Error occured while parsing file.", err);
-        return false;
+        return ERROR_FILE_PARSING;
       }
       return stored;
     }
 
     this.on("input", async function (msg) {
-      var input_vectors = JSON.parse(msg.payload);
-      var stored_vectors = await getStoredVector();
+      let input_vectors = msg.payload;
+      let stored_vectors = await getStoredVector();
```

Comment:
If this node get stored_vectors when it deployed, it can not get new face vectors that stored after flow deployed. So I think I don't have to change it.

Reply:
I agree with Jaea. I think it would be better to keep this code as it is.